File: backend/backend/src/main/java/com/moovie/backend/domain/usecase/profile/UpdateUserProfile.java
package com.moovie.backend.domain.usecase.profile;

import com.moovie.backend.domain.model.UserProfile;
import com.moovie.backend.domain.repository.ProfileRepository;
import org.springframework.stereotype.Service;

@Service
public class UpdateUserProfile {

    private final ProfileRepository profileRepository;

    public UpdateUserProfile(ProfileRepository profileRepository) {
        this.profileRepository = profileRepository;
    }

    public void invoke(UserProfile profile) {
        profileRepository.updateUserProfile(profile);
    }
}

File: src/main/kotlin/org/example/domain/usecase/profile/UpdateUserProfile.kt
package org.example.domain.usecase.profile

import kotlinx.coroutines.runBlocking
import org.example.domain.model.UserProfile
import org.example.domain.repository.ProfileRepository

class UpdateUserProfile(private val repository: ProfileRepository) {
    operator fun invoke(profile: UserProfile) = runBlocking {
        repository.updateUserProfile(profile)
    }
}
